### DISCUSSION TITLE
Using pvlib.irradiance.get_total_irradiance() for Even-Span Roofs

### DISCUSSION THREAD

**Discussion by atzounis** [pvlib/pvlib-python#1775] [Q&A]
Hello, I am investigating the use of Pvlib in a tool to calculate the actual radiation entering a greenhouse. The challenges here are multiple but the first I come through is the following:
Let's take the simplest case of an even span roof of 30° inclination on each side and a specific orientation, which can be configured through the surface azimuth. I can use the `pvlib.irradiance.get_total_irradiance()` to calculate the radiation on one side of a roof. What I am trying to figure out is a way to calculate the direct solar radiation (`poa_direct`) on the other half of the roof. I am thinking of using the surface azimuth and combining it with the solar zenith of the given day so that I can calculate the proportion of the sun's trajectory that will face each side of the roof. I am not sure if that's right though and/or if there is a less complex way to do this.
[Here is a link](https://snipboard.io/lMTKUu.jpg) to a visualization of what I am trying to explain above.
Any idea would be highly appreciated. Thank you in advance for your thoughts.

**Comment by kandersolar** [comment]
> I can use the pvlib.irradiance.get_total_irradiance() to calculate the radiation on one side of a roof. What I am trying to figure out is a way to calculate the direct solar radiation (poa_direct) on the other half of the roof. 

Can't you simply do another call to `get_total_irradiance()` but using the orientation angles for the second roof plane?  Note that `get_total_irradiance` returns both the total in-plane irradiance (`poa_global`) and its in-plane components (e.g. `poa_direct`).

**Reply by atzounis** [reply]
Thanks for your response. That could do the trick. Run 2 instances of `get_total_irradiance()` and keep only the positive values. Thanks! :)
